zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -0,0 +1,172 @@
+use ahash::RandomState;
+use std::collections::HashMap;
+use zenoh_config::{
+    AclConfig, Action, ConfigRule, Permission, PolicyRule, Subject, NUMBER_OF_ACTIONS,
+    NUMBER_OF_PERMISSIONS,
+};
+use zenoh_keyexpr::keyexpr;
+use zenoh_keyexpr::keyexpr_tree::{IKeyExprTree, IKeyExprTreeMut, KeBoxTree};
+use zenoh_result::ZResult;
+
+pub struct PolicyForSubject(Vec<Vec<KeTreeRule>>); //vec of actions over vec of permission for tree of ke for this
```

Comment:
I think this `struct` is badly defined and hard to use in the code due to some weird/undocumented index access.